### ISSUE TITLE
Emit meaningful error message instead of IndexOutOfBoundsException when demarshalling FileDescriptor without prerequisites satisfied

### ISSUE THREAD

**Issue by Flowdalic** [hypfvieh/dbus-java#230]
Invoking logind `Inhibit()`, which returns a `FileDescriptor` causes the following exception to be thrown using dbus-java 4.3.0:

```
Exception in thread "main" java.lang.IndexOutOfBoundsException: Index 0 out of bounds for length 0
        at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
        at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
        at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266)
        at java.base/java.util.Objects.checkIndex(Objects.java:361)
        at java.base/java.util.ArrayList.get(ArrayList.java:427)
        at org.freedesktop.dbus.messages.Message.extractOne(Message.java:1052)
        at org.freedesktop.dbus.messages.Message.extract(Message.java:1328)
        at org.freedesktop.dbus.messages.Message.extract(Message.java:1319)
        at org.freedesktop.dbus.messages.Message.extract(Message.java:1301)
        at org.freedesktop.dbus.messages.Message.getParameters(Message.java:1436)
        at org.freedesktop.dbus.RemoteInvocationHandler.executeRemoteMethod(RemoteInvocationHandler.java:210)
        at org.freedesktop.dbus.RemoteInvocationHandler.invoke(RemoteInvocationHandler.java:89)
        at jdk.proxy2/jdk.proxy2.$Proxy27.Inhibit(Unknown Source)
        at eu.geekplace.sandmann.Logind.$anonfun$4(Logind.scala:132)
```

**Comment by hypfvieh** [comment]
Sorry, this is by far not enough information to help in any way.
Do you have the https://github.com/rm5248/dbus-java-nativefd in your classpath?

Can you reproduce this with a minimum example and provide the required code?

**Comment by hypfvieh** [comment]
FileDescriptor only work when using jnr-transport and https://github.com/rm5248/dbus-java-nativefd as third party library.
Support for FileDescriptor without the 3rd party library will be available with version 4.3.1 when using junixsocket transport.

The error message says that you have added 2 different unix transports (jnr and native). You can only use one of them.
This is because dbus-java-nativefd pulls in jnr-transport while you also added native-transport as dependency.
Please remove native-transport from your dependencies, add dbus-java-nativefd and jnr-transport instead (please add both to be sure you use the same version for dbus-java-core and jnr-transport).

Enabling trace logging is nothing slf4j related but log4j configuration in your case. Slf4j is just a abstraction to allow using any compatible logging framework instead of enforcing one. Configuring log4j to also log trace messages should do the trick.
